jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2058,10 +2079,20 @@ ecma_op_object_get_property_names (ecma_object_t *obj_p, /**< object */
 
       if (obj_is_builtin)
       {
-        ecma_builtin_list_lazy_property_names (obj_p,
-                                               opts,
-                                               prop_names_p,
-                                               skipped_non_enumerable_p);
+        if (type == ECMA_OBJECT_TYPE_FUNCTION && ecma_builtin_function_is_routine (obj_p))
+        {
+          ecma_builtin_function_list_lazy_property_names (obj_p,
+                                                          opts,
+                                                          prop_names_p,
+                                                          skipped_non_enumerable_p);
+        }
+        else
+        {
+          ecma_builtin_list_lazy_property_names (obj_p,
```

Comment:
What about keeping this single function, and do this check at the beginnig of the function:
`if (type == ECMA_OBJECT_TYPE_FUNCTION && ecma_builtin_function_is_routine (obj_p))`

Reply:
IMHO builtin routine is a function property of builtin object (e.g. Math.abs), so it should be handled independently from builtin objects.
So, I added `ecma_builtin_routine_list_lazy_property_names` for builtin routines to clearly separate each builtin routine from builtin object.
This separation also follows other use case in which `ecma_op_function_list_lazy_property_names` and `ecma_op_function_try_to_lazy_instantiate_property` exist for function object.